src/Microsoft.AspNet.Mvc.Core/Resources.resx

```diff
@@ -466,4 +466,7 @@
   <data name="CacheProfileNotFound" xml:space="preserve">
     <value>The '{0}' cache profile is not defined.</value>
   </data>
+  <data name="ModelType_WrongType" xml:space="preserve">
+    <value>The model type '{0}' does not match the '{1}' type parameter.</value>
```

Comment:
A few notes here:
1). Is `{1}` a type name or the name of the parameter? wording makes it seem like it's the parameter name, which isn't especially helpful. (ArgumentException already provides that).
2). 'match' isn't meaningful here because it doesn't explain what kind of match you're looking for (exact, conversion, assignable). Try to be more specific.